FeedStoreChallenge/CoreData/ManagedCache.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  ManagedCache+CoreDataClass.swift
+//  FeedStoreChallenge
+//
+//  Created by Saravanakumar S on 20/06/21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+//
+
+import Foundation
+import CoreData
+
+@objc(ManagedCache)
+final class ManagedCache: NSManagedObject {
+	@NSManaged public var timestamp: Date
+	@NSManaged public var feed: NSOrderedSet
+}
+
+extension ManagedCache {
+	static func uniqueFeedCache(from context: NSManagedObjectContext) throws -> ManagedCache {
+		try fetch(context: context).map(context.delete)
+		return ManagedCache(context: context)
+	}
+
+	static func fetch(context: NSManagedObjectContext) throws -> ManagedCache? {
+		let request = NSFetchRequest<ManagedCache>(entityName: ManagedCache.entity().name!)
+		return try context.fetch(request).first
```

Comment:
To slightly increase CoreData performance `request.returnsObjectsAsFaults` can be set to `false`.

According to the [documentation](https://developer.apple.com/documentation/coredata/nsfetchrequest/1506756-returnsobjectsasfaults)
> When you execute a fetch, by default `returnsObjectsAsFaults` is `true`; Core Data fetches the object data for the matching records, fills the row cache with the information, and returns managed object as faults. These faults are managed objects, but all of their property data resides in the row cache until the fault is fired. When the fault is fired, Core Data retrieves the data from the row cache. ... If you need to access the property values from the returned objects (for example, if you iterate over all the objects to calculate the average value of a particular attribute), then it is more efficient to set `returnsObjectsAsFaults` to `false` to avoid the additional overhead.

```suggestion
		request.returnsObjectsAsFaults = false
		return try context.fetch(request).first
```

Reply:
Its very clear now. 👍